### DISCUSSION TITLE
Call smart contract function

### DISCUSSION THREAD

**Discussion by duybuikardiachain** [WalletConnect/walletconnect-monorepo#2819] [Developer Support]
I'm using Android SDK
On WC1, I'm able to call smart contract function, but on V2 seems I can't call a smart contract function on Sign Request
For example I have a function like "mintNFT"
If I put it to a method (replace eth_sendTransaction to mintNFT), it will show an error

**Comment by finessevanes** [comment]
Details such as the following would help:
1) Version of the package you are using
2) Reproducible code via a codeBox, error logs, and/or reproduction steps

**Reply by thanhduy26091995** [reply]
1. I used latest version of Metamask
2. Here is an error: JsonRpcError(id=1687960879755529, code=-32603, message=Internal error)
Request Params capture
![Capture](https://github.com/WalletConnect/walletconnect-monorepo/assets/18645098/fc770307-5c14-4ca1-b50e-4c268c66da77)

**Reply by duybuikardiachain** [reply]
Please help to check it, it lock our migration from V1 to V2
